api/src/main/java/javax/el/ELUtil.java

```diff
@@ -222,13 +222,22 @@ static Method findMethod(Class<?> klass, String methodName, Class<?>[] paramType
      */
     static Object invokeMethod(ELContext context, Method method, Object base, Object[] params) {
 
-        Object[] parameters = buildParameters(context, method.getParameterTypes(), method.isVarArgs(), params);
         try {
+            int paramCount = params == null ? 0 : params.length;
+            if (paramCount == 0) {
+                for (Class ParameterType : method.getParameterTypes()) {
+                    if (ParameterType.getName().equals("javax.faces.event.AjaxBehaviorEvent")) {
+                        return method.invoke(base, (Object[]) params);
+                    }
+                }
+            }
+            Object[] parameters = buildParameters(
+                    context, method.getParameterTypes(), method.isVarArgs(), params);
             return method.invoke(base, parameters);
         } catch (IllegalAccessException iae) {
             throw new ELException(iae);
         } catch (IllegalArgumentException iae) {
-            throw new ELException(iae);
+            throw new IllegalArgumentException(iae);
```

Comment:
you can rethrow iae directly. But ... to me it looks like it was explicitly desired to throw ELException instead.
I think that in good code this catch block should not be there. But first we need to know why is it here.

Reply:
I'm reviewing it for second time and I think the ElException is desired here, so it  should be reset back or explained at least.